src/ui/pages/SwapForm/SwapForm.tsx

```diff
@@ -519,11 +535,69 @@ export function SwapForm() {
                 chain={chain}
                 transaction={configureTransactionToBeSigned(currentTransaction)}
                 configuration={swapView.store.configuration.getState()}
+                onOpenAllowanceForm={() =>
+                  allowanceDialogRef.current?.showModal()
+                }
               />
             </ViewLoadingSuspense>
           );
         }}
-      ></BottomSheetDialog>
+      />
+
+      <BottomSheetDialog
+        ref={allowanceDialogRef}
+        key="allowance"
```

Comment:
what for?

Reply:
I don't think it is necessary here, added for consistency with the `BottomSheetDialog` above, I'll delete this key